mypy/server/update.py

```diff
@@ -594,7 +594,16 @@ def calculate_active_triggers(manager: BuildManager,
             names.add(id)
         else:
             snapshot2 = snapshot_symbol_table(id, new.names)
-        names |= compare_symbol_table_snapshots(id, snapshot1, snapshot2)
+        diff = compare_symbol_table_snapshots(id, snapshot1, snapshot2)
+        package_nesting_level = id.count('.')
+        for item in diff:
+            if (item.count('.') <= package_nesting_level + 1
+                    and not item.split('.')[-1].startswith('_')):
```

Comment:
Shouldn't we treat `.pyi` files specially here? For example ignore top level names imported from other modules (unless imported using `... as ...`).

Reply:
That would improve efficiency a bit but since `.pyi` files are usually shipped as part of typeshed, they don't change very frequently and this is a minor problem. Added a comment about this.